### DISCUSSION TITLE
Display cells with particles inside

### DISCUSSION THREAD

**Discussion by gmsafir** [firemodels/fds#13212] [General]
I have a case of vegetation defined as particles in a cone : 

&INIT ID='init_foliage', 			PART_ID='part_foliage', 			XYZ=0.0,0.0,0.15, RADIUS=0.35, HEIGHT=2.5, SHAPE='CONE', N_PARTICLES_PER_CELL=1, CELL_CENTERED=.TRUE., MASS_PER_VOLUME=2.82 /

Is there a way to display/highlight in green color somehow in SMV the mesh cells with particles inside?

I have tried without success:
&ISOF QUANTITY='PARTICLE TEMPERATURE', VALUE=50 /
&ISOF QUANTITY='PARTICLE MASS', VALUE=1E-5 /
(it does run but providing no isosurfaces)
as well as:
&ISOF QUANTITY='PARTICLE TEMPERATURE', PART_ID='part_foliage', 	VALUE=50 /
(produces an error)

**Comment by ericvmueller** [comment]
Since you have one particle per cell, placed at the center, perhaps you can try using a `PROP` line to render the particles as boxes which are equivalent to the cell size. Check out section 18.7.3 in the user guide

![image](https://github.com/user-attachments/assets/1c204e36-64d5-4a83-9118-6a47bffacfba)

**Comment by gmsafir** [comment]
Thanks Eric, I tried with FDS6.9.1 to use the new PROP line, that way:
&HEAD CHID='test_smv_id', TITLE='test SMV id' /

&MESH ID='Mesh-a-a', IJK=16,16,30, XB=-2.00,+2.00,-2.00,+2.00,00.0,7.50/

&TIME T_END=120 / 

&DUMP
DT_HRR	=1
DT_DEVC	=1
DT_PART	=1
DT_SLCF	=1 /

&VENT MB='XMIN', SURF_ID='OPEN' /
&VENT MB='XMAX', SURF_ID='OPEN' /
&VENT MB='YMIN', SURF_ID='OPEN' /
&VENT MB='YMAX', SURF_ID='OPEN' /
&VENT MB='ZMAX', SURF_ID='OPEN' /
&VENT PBZ=0,    SURF_ID='surf_foil' /

&SURF ID='surf_foil'
      COLOR = 'SILVER'
      MATL_ID='matl_alu','matl_beton'
      THICKNESS(1:2) = 0.0002,0.01 /

&MATL ID='matl_alu'
      DENSITY = 2707.
      SPECIFIC_HEAT = 0.896
      CONDUCTIVITY = 204.
      EMISSIVITY = 0.1 /

&MATL ID='matl_beton'
      DENSITY = 1500.
      SPECIFIC_HEAT = 1.
      CONDUCTIVITY = 0.3  /
	  

&SURF
ID                      	= 'surf_50pc_foliage'
MATL_ID(1,1:2)                          = 'matl_wood','matl_moisture'
MATL_MASS_FRACTION(1,1:2)	= 0.8772, 0.1228
SURFACE_VOLUME_RATIO   		= 3940
LENGTH                 		= 0.05
GEOMETRY               		= 'CYLINDRICAL'

/

&PROP
ID = 'prop_foliage'
SMOKEVIEW_ID='box'
SMOKEVIEW_PARAMETERS=’R=192’,’G=255’,’B=128’,’DX=0.25’,’DY=.25’,’DZ=0.25’
/

&PART
ID				= 'part_foliage'
PROP_ID = 'prop_foliage'
SAMPLING_FACTOR	= 1
SURF_ID			= 'surf_50pc_foliage'
QUANTITIES		= 'PARTICLE TEMPERATURE','PARTICLE MASS',
STATIC			= .TRUE.,
COLOR			= 'FOREST GREEN' /

&INIT ID='init_foliage', 			PART_ID='part_foliage', 			XYZ=0.0,0.0,0.15, RADIUS=0.35, HEIGHT=2.5, SHAPE='CONE', N_PARTICLES_PER_CELL=1, CELL_CENTERED=.TRUE., MASS_PER_VOLUME=5.25 /


&REAC
FUEL='WOOD',
SOOT_YIELD= 0.010
CO_YIELD= 0.0040
HEAT_OF_COMBUSTION= 17.5E3/

&SPEC ID='WOOD', FORMULA= 'C3.4H6.2O2.5'/

&SPEC ID='WATER VAPOR' /
&SPEC ID='WATER VAPOR MOISTURE'   SPEC_ID='WATER VAPOR'/

&MATL
ID           		= 'matl_moisture'
DENSITY      		= 1000
CONDUCTIVITY 		= 0.6
SPECIFIC_HEAT 		= 4.190
/


&MATL
ID 					= 'matl_wood'
DENSITY 			= 514
CONDUCTIVITY 		= 0.11
SPECIFIC_HEAT_RAMP 	= 'cp_dry_veg'
/

&RAMP ID='cp_dry_veg', T= 0.,    F=1.18 /
&RAMP ID='cp_dry_veg', T= 1000., F=4.81 /


&TAIL /

But I get an error:
ERROR(101): Problem with PROP number 1, line number 47 (CHID: test_smv_id)


Any idea?

**Reply by drjfloyd** [reply]
You appear to be using two differnt quote marks. This could just be because of how you pasted your text in, but make sure all the inputs have the single vertical quote mark and not the tilted quote mark.

**Reply by gmsafir** [reply]
Thanks, a new progress!!! Now FDS-6.9.1 is running correctly. However, when opening the SMV file with SMV-6.9.1, I don't see the particles rendered as boxes, only as classical point particles

**Reply by mcgratta** [reply]
In Smokeview, Show/Hide -> Particles -> Draw Using ...

**Comment by gmsafir** [comment]
Not found in my Windows version of SMV-6.9.1...
![image](https://github.com/user-attachments/assets/19c560f5-6c38-4e64-a254-9058011be23d)

**Reply by mcgratta** [reply]
Follow the arrow after `part_foliage`

**Reply by gmsafir** [reply]
just activates or deactivate the same parts rendered as points
![image2](https://github.com/user-attachments/assets/051b1bc0-dd3b-4d4d-a493-f0c74e201832)

**Reply by mcgratta** [reply]
![image](https://github.com/user-attachments/assets/c6b4c69c-5841-4c54-943d-8bbad567865c)

**Comment by gmsafir** [comment]
also tried with a Linux version SMV-6.9.1 (installed with binaries, not from sources) on WSL-Ubuntu, accessed via WSL-window or via MobaXterm, but got 2 different errors:
- with WSL-Ubuntu: reading  tree_2_m_14_pc.smv
  pass 1
  pass 2
  pass 3
  pass 4
  wrapping up
complete

corrupted double-linked list
Aborted


- with MobaXterm on WSL-Ubuntu: 
  Smokeview is running on a system using OpenGL 1.4 (2.1 Mesa 10.5.4)
   OpenGL 2.0 or later is required to use the GPU.
   GPU smoke shader not supported.
  GPU shader initialization failed
reading  tree_2_m_14_pc.smv
  pass 1
  pass 2
  pass 3
  pass 4
  wrapping up
complete


Startup time: 1.4 s

research mode on, using global bounds

X connection to localhost:10.0 broken (explicit kill or server shutdown).

**Comment by gmsafir** [comment]
it now works! For unexpected reason, in the SMV file produced by FDS6.9.1, the PROP "prop_foliage" was well defined, but one line was produced incomplete, namely:
CLASS_OF_PARTICLES
 part_foliage                                            
      0.13333      0.54510      0.13333

instead of:

CLASS_OF_PARTICLES
 part_foliage                                                 % % prop_foliage                                                
      0.13333      0.54510      0.13333

**Comment by gmsafir** [comment]
last question: with following box definition, why is it that the box is rendered 2D instead of 3D ?

PROP
 prop_foliage
  1
 box
  6
 R=192                                                      
 G=255                                                       
 B=128                                                       
 DX=0.12                                                       
 DY=0.12                                                     
 DZ=0.12                                                     

![image4](https://github.com/user-attachments/assets/578aa98a-9eb5-461d-9bf7-5d838c537158)

**Comment by gmsafir** [comment]
I use FDS-6.9.1-0-g889da6a-release

**Comment by mcgratta** [comment]
[test_smv_id.txt](https://github.com/user-attachments/files/16367775/test_smv_id.txt)

Glenn -- can you run this case a few time steps and check if the box is drawn correctly.

**Comment by gmsafir** [comment]
Hi, any update please, Glenn?

**Comment by gforney** [comment]
should be fixed in nightly bundle posted here

https://github.com/firemodels/test_bundles/releases/tag/FDS_TEST

**Comment by gmsafir** [comment]
Glenn, sorry but still a problem with the nightly bundle.

Left is with head version of 6.9.1, right with the nightly bundle:
![image1](https://github.com/user-attachments/assets/f069b070-f5dc-40cc-a51d-5407c696b543)
although I have a prop foliage defined as:
PROP
 prop_foliage
  1
 box
  6
 R=0                                                      
 G=255                                                       
 B=0                                                   
 DZ=0.12
 DX=0.12                                                       
 DY=0.12  

Here the complete smv file: 
[tree_2_m_14_pc.txt](https://github.com/user-attachments/files/16602142/tree_2_m_14_pc.txt)

**Comment by gforney** [comment]
Kevin do you still see the problem?

**Reply by mcgratta** [reply]
I see a green cube or a blue X depending on where the .svo file is.

**Comment by gforney** [comment]
the particles are small but are drawn as a cube - no longer flat.  what were you expecting or is there some other problem?

![test_smv_id_0000](https://github.com/user-attachments/assets/0c8a7f3e-74ce-43e8-83ea-841e8b3079e3)

**Comment by mcgratta** [comment]
I see this
![image](https://github.com/user-attachments/assets/f0b0400d-4071-4099-9637-b155b7d5eeac)

**Reply by mcgratta** [reply]
This picture is correct. The grid cells are 0.25 m on a side, and the property of the particles is to draw a green box 0.25 m on a side. The particles are in the center of each cell, and therefore we get a solid green box in total.

**Comment by gmsafir** [comment]
Great that is the correct output that I expect, with the green boxes around the particles occupying the entire cells! Kevin, would it possible to get the svo file that you taylored to make it work ? That would be of great help to me, to understand the dataflow btw smv and svo flies, absolutely not clear to me (I thought the smv would contain all infos about particle props)

**Comment by gforney** [comment]
The .svo file is in the directory where you installed smokeview or in your
smv repo at smv\Build\for_bundle. If you were running smokeview that you
compuled you need to add a -bindir parameter to your smokeview command
line. Something like
smokeview_exe_name -bindir path_to_for_bundle casename

for_bundle is in under smv\Build

**Comment by gforney** [comment]
I reproduced the problem, sorry for the delay. I'm investigating .  in the mean time try the -bindir parameter to point to the for_bundle directory or try copying the .svo file from this directory to the directory containing your case

**Comment by gforney** [comment]
two locations where the file objects.svo are located on a windows computer
C:\Program Files\firemodels\SMV6
path_to_smv_repo\smv\Build\for_bundle

On Wed, Aug 14, 2024 at 7:15 AM gforney ***@***.***> wrote:

> I reproduced the problem, sorry for the delay. I'm investigating . in the
> mean time try the -bindir parameter to point to the for_bundle directory or
> try copying the .svo file from this directory to the directory containing
> your case
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/firemodels/fds/discussions/13212#discussioncomment-10336241>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC6UCRXGVH5V3OIPH3CMQ73ZRM35PAVCNFSM6AAAAABLMENSP2VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTAMZTGYZDIMI>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Glenn Forney

**Comment by gmsafir** [comment]
with -bindir, it helps to change the display from blue crosses to green boxes, but they are not yet at full cell size.
I use the following command: full_path_smokeview.exe -bindir path_to_dir_of_my_svo_file tree_2_m_14_pc.smv


![image1_smv](https://github.com/user-attachments/assets/aba6a994-ee24-4fa8-baa5-9b8f3cc63251)
![image2_smv](https://github.com/user-attachments/assets/056428bc-a5eb-41cc-83a6-231ea0ad3bd5)

**Comment by gmsafir** [comment]
(smv is the same as previously sent)

**Comment by gmsafir** [comment]
Thanks Kevin for your sample. I will run it and cross-compare with my smv, to see what is wrong in mine

**Comment by gmsafir** [comment]
Glenn, I did run with FDS-6.9.1 (head) the case by Kevin, then used your last nightly SMV-6.9.1 with correction and I still have the little boxes

![image3_smv](https://github.com/user-attachments/assets/0e9c900d-6373-47da-b60f-650267c28a05)

Do I need to run the last nightly FDS-6.9.1 ? where do I get the source code to recompile on my cluster? (install with .sh file often fails)

**Comment by gmsafir** [comment]
Hi Glenn, any fededback?

**Comment by gforney** [comment]
until I post a fix do the following
1. copy objects.svo.txt (attached) to the directory containing your case and change its name to object.svo
2. if you want to change the size of your boxes change the DX DY and DZ parameters in the following line in your input file
SMOKEVIEW_PARAMETERS='R=192','G=255','B=128','DX=1','DY=1','DZ=1'

[objects.svo.txt](https://github.com/user-attachments/files/16679681/objects.svo.txt)

**Comment by gmsafir** [comment]
Glenn, good progress, but I don't understand exactly the logic.
I get the correct output (particle displayed as a box occupying the entire cell) wihen imposing DX=2.0,DY=2.0,DZ=2.0, though my grid is of resolution 0.5x0.5x0.5.
This result is obtained either with objects.svo in the case directory, or with -bindir smv6_exe_dir.
![smv_image](https://github.com/user-attachments/assets/f4ae4bfc-a30d-4def-a639-3b53f9d98cc9)

**Comment by gforney** [comment]
not sure I understand your comment- this line only affects the visual display it has nothing to do with the grid resolution.  are you saying that setting the parameters to 2 gave you a different size than what you were expecting?
SMOKEVIEW_PARAMETERS='R=192','G=255','B=128','DX=1','DY=1','DZ=1'

using -bindir giving the same results as objects.svo in your case directory is good - that means that -bindir is working

**Comment by gmsafir** [comment]
I did put these parameters to get it work : SMOKEVIEW_PARAMETERS='R=192','G=255','B=128','DX=2','DY=2','DZ=2'
I thought intuitively that DX,DY,DZ in this line would have to be connected to the resolution of my FDS grid which is 0.5x0.5x0.5
Hope this is a clearer explanation

**Comment by gforney** [comment]
the dx, dy and dz parameters on this line tell smokeview how big to make
the object.  they are nothing to do with the grid resolution

On Tue, Aug 20, 2024 at 4:04 PM gmsafir ***@***.***> wrote:

> I did put these parameters to get it work :
> SMOKEVIEW_PARAMETERS='R=192','G=255','B=128','DX=2','DY=2','DZ=2'
> I thought intuitively that DX,DY,DZ in this line would have to be
> connected to the resolution of my FDS grid which is 0.5x0.5x0.5
> Hope this is a clearer explanation
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/firemodels/fds/discussions/13212#discussioncomment-10399936>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC6UCRUWA5MSG457HBPY5STZSOOMPAVCNFSM6AAAAABLMENSP2VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTAMZZHE4TGNQ>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Glenn Forney

**Comment by gmsafir** [comment]
ok, anyhow thnaks for this great functionality to display the particles! now it works perfectly for me
